saleor/discount/migrations/tasks/saleor3_16.py

```diff
@@ -294,8 +296,10 @@ def migrate_translations(sale_ids, saleid_promotion_map):
 @app.task
 def migrate_checkout_line_discounts_task(sale_ids, rule_id_by_channel_sale):
     checkout_line_discounts = CheckoutLineDiscount.objects.filter(
-        sale_id__in=sale_ids
-    ).exclude(type="promotion")[:CHECKOUT_LINE_DISCOUNT_BATCH_SIZE]
+        Exists(Checkout.objects.filter(token=OuterRef("line__checkout")))
+        & ~Q(type="promotion"),
+        sale_id__in=sale_ids,
+    )[:CHECKOUT_LINE_DISCOUNT_BATCH_SIZE]
```

Comment:
Wouldn't this work?
```suggestion
        sale_id__in=sale_ids,
        Exists(Checkout.objects.filter(token=OuterRef("line__checkout")))
    )
    .exclude(type="promotion")[:CHECKOUT_LINE_DISCOUNT_BATCH_SIZE]
```
would be a bit cleaner.